File: lectures/14/Books/app/src/main/java/ro/ubbcluj/cs/books/service/BookService.java
package ro.ubbcluj.cs.books.service;

import java.util.List;

import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import ro.ubbcluj.cs.books.domain.Book;
import rx.Observable;

public interface BookService {
  String SERVICE_ENDPOINT = "http://192.168.2.1:3000";


  @GET("books")
  Observable<List<Book>> getBooks();

  @POST("book")
  Observable<Book> addBook(@Body Book e);

}

File: archives/2018-2019/13/Books/app/src/main/java/ro/ubbcluj/cs/books/service/BookService.kt
package ro.ubbcluj.cs.books.service

import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import ro.ubbcluj.cs.books.domain.Book
import rx.Observable

interface BookService {

  @get:GET("books")
  val books: Observable<List<Book>>

  @POST("book")
  fun addBook(@Body e: Book): Observable<Book>

  companion object {
    const val SERVICE_ENDPOINT = "http://10.0.2.2:3000"
  }
}
